compiler/rustc_mir/src/transform/const_prop.rs

```diff
@@ -418,11 +421,27 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         }
     }
 
+    /// Some `InterpError`s could be ignored but must not be to ensure that queries are stable.
+    fn check_interpresult<T>(tcx: TyCtxt<'tcx>, error: &InterpResult<'tcx, T>) {
+        if let Err(e) = error {
+            if matches!(
+                e.kind(),
+                InterpError::ResourceExhaustion(ResourceExhaustionInfo::MemoryExhausted)
+            ) {
+                // Memory errors can't be ignored since otherwise the amount of available
+                // memory influences the result of optimization and the build. The error
+                // doesn't need to be fatal since no code will actually be generated anyways.
+                tcx.sess.fatal("memory exhausted during optimization");
```

Comment:
We have all this huge error reporting machinery for InterpError, please don't do ad-hoc error reporting here.